core/src/main/scala/kafka/server/FetchSession.scala

```diff
@@ -35,7 +35,7 @@ import scala.math.Ordered.orderingToOrdered
 import scala.collection.{mutable, _}
 import scala.collection.JavaConverters._
 
-object FetchSession {
+object FetchSession extends Logging {
```

Comment:
It seems this object has no need to extend from Logging.

Reply:
I'm putting it here so all subclasses can use the same logger instance and so I don't need to create a large number of companion objects.  What's the scala (or Kafka) best practice in this case?

Reply:
Not sure what you mean. This is an object, and thus it doesn't have subclasses. Can we simply not change it and use the previous code?

Reply:
This is my understanding which could be wrong since I may be understanding trait inheritance in scala correctly.  Previously if I wanted to configure log4j2 to log everything that was subclassed from FetchSession I could add a config to log4j2 like:
```xml
  <Loggers>
        <Logger name="kafka.server.FetchContext" level="debug" additivity="true">
            <appender-ref ref="fileLogger" level="debug" />
        </Logger>
       <!-- other loggers -->
</Loggers>

```
The alternative would be to make a change to all the subclasses that use Logging.   But the log4j2 configs for that would be more complicated because each subclass would need to be named in the configuration rather than just kafka.server.FetchContext